### PR TITLE
Fix legend color to be in sync with plot color on change

### LINKED ISSUE #5233: Legend color is not in sync with plot color on change
#### Summary
Legend needs to update swatch color when user sets color from Plot Series area in Inspector.

#### Expected vs Current Behavior
User currently must nav away and back to see the change, which is undesirable.

#### Steps to Reproduce
1. Create an overlay plot
2. Add a telemetry endpoint to it
3. Change the color of the series from inspector properties
4. Observe that the plot series color changes, but the legend shows the old color in the swatch.

#### Environment
* Open MCT Version: 2.0.3
* Deployment Type: all
* OS: Mac / Windows
* Browser: All

#### Impact Check List

- [x] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [x] Is there a workaround available?
- [ ] Does this impact a critical component?
- [x] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?

#### Additional Information
In the image below, the swatch colors don't match the plot colors.

image

### PR DESCRIPTION
**Pull request by alizenguyen**
Closes #5233 

### Describe your changes:

Added observer to watch domain object changes to the series color path. Pulled and mimicked functionality regarding the series path from seriesForm.vue component where the color change is being handled.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/plugins/plot/legend/PlotLegendItemCollapsed.vue [hunk 1]
```diff
@@ -83,7 +84,8 @@ export default {
             formattedYValue: '',
             formattedXValue: '',
             mctLimitStateClass: '',
-            formattedYValueFromStats: ''
+            formattedYValueFromStats: '',
+            colorObserver: null
```
Comment (shefalijoshi) [inline src/plugins/plot/legend/PlotLegendItemCollapsed.vue:88]: colorObserver doesn't need to be defined here. Let's remove it.
Also, anything within the data's return {} becomes reactive, so unless it is being used by a computed property OR by the template above, we should avoid adding variables here.

**Review by shefalijoshi — CHANGES_REQUESTED** [review]

File: src/plugins/plot/legend/PlotLegendItemCollapsed.vue (hunk 1 above)
Comment (alizenguyen) [reply src/plugins/plot/legend/PlotLegendItemCollapsed.vue]: Removed! Thanks for reviewing!

**Comment by shefalijoshi** [comment]
@alizenguyen IT looks like this PR's target is master, but the issue is meant for 2.0.6.
Can you create a new PR for release/2.0.6 since it looks like this branch has been updated with changes from master?

**Comment by alizenguyen** [comment]
> 

Ah, yes mam! I'll recreate a PR later today. Thank you!

**Pull request closed without merging**